lib/smppex/mc/smpp_handler.ex

```diff
@@ -21,32 +21,16 @@ defimpl SMPPEX.SMPPHandler, for: SMPPEX.MC.SMPPHandler do
 
   require Logger
 
-  def after_init(_session) do
-  end
-
-  def handle_parse_error(session, error) do
-    Logger.info("mc_conn #{inspect session.mc_conn}, parse error: #{inspect error}, stopping")
-  end
-
-  def handle_pdu(session, {:unparsed_pdu, raw_pdu, error}) do
-    Logger.info("mc_conn #{inspect session.mc_conn}, unknown pdu: #{inspect raw_pdu}(#{inspect error}), stopping")
-    :stop
+  def handle_pdu(_session, {:unparsed_pdu, raw_pdu, error}) do
+    {:stop, {:parse_error, {:unparsed_pdu, raw_pdu, error}}}
   end
 
   def handle_pdu(session, {:pdu, pdu}) do
     :ok = MC.handle_pdu(session.mc_conn, pdu)
   end
 
-  def handle_socket_closed(session) do
```

Comment:
I like how this cleans up the extra function signatures, and also makes it possible to silence these logs! 👍